package Message; 
message CombatReport								//战报
{
   message Action
   {
      message CheckAttr
      {
	optional int32 idx = 1;		//位置
	optional int32 attr = 2;	//相关属性
	optional int32 value = 3;	//属性值
      }

	optional int32		round 	      = 1;	//回合
	optional int32		attackerindex = 2;	//攻击者
	optional int32		skill	      = 3;	//使用技能
	repeated int32		defenderindex = 4;	//被攻击者
	repeated int32		damage 	      = 5;	//伤害
	repeated int32		flag 	      = 6;	//其它标记
	repeated int32		counter	      = 7;	//反击伤害
	repeated int32		buff	      = 8;	//buff
	optional int32		combo	      = 9;	//连击数
	repeated CheckAttr	checkattr     = 10;	//属性校验
   }
   
   repeated Action		action		= 1;	//每次动作序
} 

